youtube_dl/downloader/__init__.py

```diff
@@ -30,6 +30,10 @@
 def get_suitable_downloader(info_dict, params={}):
     """Get the downloader class that can handle the info dict."""
     protocol = determine_protocol(info_dict)
+    if protocol == 'm3u8' and params.get('hls_prefer_native') is True:
+        protocol = 'm3u8_native'
+    elif protocol == 'm3u8_native' and params.get('hls_prefer_native') is False:
+        protocol = 'm3u8'
```

Comment:
Ok, so if you don't want to do it this way,
Could you add `m3u8_native` in the [external.py#L227](https://github.com/rg3/youtube-dl/blob/master/youtube_dl/downloader/external.py#L227) check because --hls-prefer-ffmpeg is currently broken.